compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1197,6 +1197,11 @@ public void visit(BLangRecordVariable varNode) {
             varNode.setBType(symResolver.resolveTypeNode(varNode.typeNode, env));
         }
 
+        if (isOptionalFieldAvailableInRecord(varNode.typeNode.getBType())) {
```

Comment:
This will disallow `{name: eName, id: eId} = e;` as well right? we need to allow this